From ea73ffab6844d1253eb66ff152954bab5542dce9 Mon Sep 17 00:00:00 2001
From: Olga Arkhangelskaia <arkholga@tarantool.org>
Date: Fri, 22 Nov 2019 14:39:22 +0300
Subject: [PATCH v2 1/2] build: enables DLUAJIT_ENABLE_PAIRSMM by default

Turns on DLUAJIT_ENABLE_PAIRSMM flag for tarantool build.
Now pairs/ipairs metamehods are available.
---
 cmake/luajit.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/cmake/luajit.cmake b/cmake/luajit.cmake
index 10df633d5..072db8269 100644
--- a/cmake/luajit.cmake
+++ b/cmake/luajit.cmake
@@ -217,6 +217,7 @@ macro(luajit_build)
         add_definitions(-DLUAJIT_USE_ASAN=1)
         set (luajit_ldflags ${luajit_ldflags} -fsanitize=address)
     endif()
+    add_definitions(-DLUAJIT_ENABLE_PAIRSMM=1)
     add_definitions(-DLUAJIT_SMART_STRINGS=1)
     # Add all COMPILE_DEFINITIONS to xcflags
     get_property(defs DIRECTORY PROPERTY COMPILE_DEFINITIONS)
-- 
2.20.1 (Apple Git-117)

